core-bundle/src/Security/Voter/DataContainer/AbstractDynamicPtableVoter.php

```diff
@@ -78,16 +88,39 @@ private function getParentTableAndId(int $id): array
 
         // Limit to a nesting level of 10
         $records = $this->connection->fetchAllAssociative(
-            "SELECT id, @pid:=pid AS pid, ptable FROM $table WHERE id=?".str_repeat(" UNION SELECT id, @pid:=pid AS pid, ptable FROM $table WHERE id=@pid", 9),
-            [$id],
+            "SELECT id, @pid:=pid AS pid, ptable FROM $table WHERE id=:id".str_repeat(" UNION SELECT id, @pid:=pid AS pid, ptable FROM $table WHERE id=@pid AND ptable=:ptable", 9),
+            ['id' => $id, 'ptable' => $table],
```

Comment:
```suggestion
            "SELECT id, @pid:=pid AS pid, ptable FROM $table WHERE id=:id AND ptable=:ptable".str_repeat(" UNION SELECT id, @pid:=pid AS pid, ptable FROM $table WHERE id=@pid AND ptable=:ptable", 9),
            ['id' => $id, 'ptable' => $table],
```

Do we not need the `AND ptable=:ptable` in the first SELECT?

Reply:
we could, but it does not make sense because the method is only called if that statement is already true.

Reply:
it would allow us to use the first record a few lines below

Reply:
You are right, the first query uses `id=:id` and not `id=@pid`, so it is not necessary to add this.